### PR TITLE
Improved code review guidelines

### PR DESCRIPTION
**Pull request by oliverguenther**

### CODE REVIEW

File: docs/development/code-review-guidelines/README.md [hunk 1]
```diff
@@ -1,56 +1,105 @@
...
+
+Try to think of edge cases when testing or evaluating the code, double check the test coverage. But do not frown if you merged the pull request and something broke after all. This is the learning path to avoiding this mistake on the next attempt. Not doing a review in the first place will not move you forward either.
+
+
 
 ## Coding style
 
-We try to adhere to the [Ruby community styleguide](https://github.com/bbatsov/ruby-style-guide). At some point we will have to make decisions about some rules that are not clear or defined within that styleguide. In that case, we should either fork it or note all decisions here.
+We try to adhere to the [Ruby community styleguide](https://github.com/bbatsov/ruby-style-guide) as well as the [AirBnB JavaScript style guide](https://github.com/airbnb/javascript) with [some extensions for Angular](../style-guide/frontend/). Rules we want to follow are expressed as either Rubocop definitions are eslint rules. Follow your linter to adhere to them.
```
Comment (klaustopher) [inline docs/development/code-review-guidelines/README.md]:
```suggestion
We try to adhere to the [Ruby community styleguide](https://github.com/bbatsov/ruby-style-guide) as well as the [AirBnB JavaScript style guide](https://github.com/airbnb/javascript) with [some extensions for Angular](../style-guide/frontend/). Rules we want to follow are expressed as either Rubocop definitions or eslint rules. Follow your linter to adhere to them.
```

File: docs/development/code-review-guidelines/README.md [hunk 2]
```diff
@@ -1,56 +1,105 @@
...
+
+Before committing, please run your new code through [Rubocop](https://github.com/bbatsov/rubocop). It detects deviations from a lot of things in the style guide and things that are bad practice in general. You obviously do not have to fix issues with existing code. There is a [list of editor plugins](https://docs.rubocop.org/rubocop/1.31/integration_with_other_tools.html#editor-integration) in the Rubocop docs. You can also use `bin/dirty-rubocop` to test them. Pull requests are being linted automatically through a GitHub action.
 
-Most of our code does not yet adhere to this styleguide, but we want all new code to adhere to it. You do not have to improve existing code when making changes, but we encourage it. If you do, please do all improvements in a separate commit from the actual change, so the improvements do not hide your actual code changes in a diff.
+The same is true for eslint. Your editor will likely have support for eslint checks, and allows you to correct them before committing.
```
Comment (klaustopher) [inline docs/development/code-review-guidelines/README.md:17]: Maybe suggest that we have lefthook available and it can be used to install the `pre-commit` hook to automatically check your code before you commit?

File: docs/development/code-review-guidelines/README.md [hunk 3]
```diff
@@ -1,56 +1,105 @@
...
-- All tests must be green on continuous integration
+- All GitHub workflow actions must be green on continuous integration
 - Appropriate unit and integration tests, e.g. test application logic via rspec tests and its integration with the user interface via Cucumber tests
-- Test JavaScript code, e.g. via Jasmine or Selenium/Cucumber, but: Cucumber tests that do not test JavaScript must not have the @javascript tag, otherwise they run slower
-- Look for sufficient coverage of both Ruby and JavaScript code. CI will provide at least Ruby line coverage information at some point in the future
+- Test Frontend code with appropriate feature specs.
 - Forms: besides testing for success response, also test whether values are actually saved, e.g. read form or look into database
 - Bugfixes: must contain a test which detects the bug they fix to prevent regressions
 - Translations: Never use a specific translation string in a test. We might want to change them in the future and do not want to fix tests when we do.
+- Manual testing: When testing a feature or change, check out the code test at least the happy paths according to the specification of the ticket.
```
Comment (klaustopher) [inline docs/development/code-review-guidelines/README.md]: Maybe a note that we are aware of flickering specs, actively trying to remove them. Something along

> - We are aware that there are some flickering specs in our codebase, that might fail randomly. We are actively trying to fix those. If you encounter test failures in code that you have not touched, try re-running the specs.

**Review by klaustopher — APPROVED** [review]

File: docs/development/code-review-guidelines/README.md (hunk 2 above)
Comment (oliverguenther) [reply docs/development/code-review-guidelines/README.md:17]: good point, I thought about that but that somehow got carried away and didn't document it after all. Will fix that

File: docs/development/code-review-guidelines/README.md [hunk 4]
```diff
@@ -60,6 +109,33 @@ The reviewer should understand the code without explanations outside the code.
 
 *You should not rush through a code review – but also, you need to do it promptly. Your coworkers are waiting for you.*
 
+
+
+## Completing the review
+
+If you have completed the review, you either have left feedback for the developer. In that case:
```
Comment (aaron-contreras) [inline docs/development/code-review-guidelines/README.md]:
```suggestion
Once you've completed the review, you might have left feedback for the developer. In that case:
```
